client/src/components/Collections/sheet/workbooks.ts

```diff
@@ -0,0 +1,108 @@
+import { GalaxyApi, type SampleSheetColumnDefinition, type SampleSheetColumnDefinitions } from "@/api";
+import type { SampleSheetCollectionType } from "@/api/datasetCollections";
+import { stripExtension } from "@/components/Collections/common/stripExtension";
+import type { PrefixColumnsType } from "@/components/Collections/wizard/types";
+import { withPrefix } from "@/utils/redirect";
+
+export function getDownloadWorkbookUrl(
+    columnDefinitions: SampleSheetColumnDefinitions,
+    collectionType: SampleSheetCollectionType,
+    initialRows?: string[][]
+) {
+    const columnDefinitionsJson = JSON.stringify(columnDefinitions);
+    const columnDefinitionsJsonBase64 = Buffer.from(columnDefinitionsJson).toString("base64");
+    let url = withPrefix(
+        `/api/sample_sheet_workbook/generate?collection_type=${collectionType}&column_definitions=${columnDefinitionsJsonBase64}`
+    );
+    if (initialRows) {
+        const initialRowsJson = JSON.stringify(initialRows);
+        const initialRowsJsonBase64 = Buffer.from(initialRowsJson).toString("base64");
+        url = `${url}&prefix_values=${initialRowsJsonBase64}`;
+    }
+    return url;
+}
+
+export function getDownloadWorkbookUrlForCollection(column_definitions: SampleSheetColumnDefinitions, hdca_id: string) {
+    const columnDefinitionsJson = JSON.stringify(column_definitions);
+    const columnDefinitionsJsonBase64 = Buffer.from(columnDefinitionsJson).toString("base64");
+    const url = withPrefix(
+        `/api/dataset_collections/${hdca_id}/sample_sheet_workbook/generate?column_definitions=${columnDefinitionsJsonBase64}`
```

Comment:
This seems odd, why not make that a normal request body ? We don't know what people's proxy's consider too long, I think the nginx default for the entire request line is 8K